Reorganize integration docs (#66)
Layr-Labs/eigenda-docs: docs/integrations-guides/eigenda-proxy.md

# EigenDA Proxy

## About

> EigenDA proxy is a sidecar server ran as part of a rollup node cluster for communication with the EigenDA network. 

Reviewer: ```suggestion
EigenDA proxy is a sidecar REST server meant to be run next to a rollup node for communicating with the EigenDA network. 
```

Reviewer: "cluster" was confusing to me, and probably will be to users. If this is meant to be an intro guide, then we should just say node<1-1>proxy

Author: [4d3d60d](https://github.com/Layr-Labs/eigenda-docs/pull/66/commits/4d3d60d271bd334a04f90f8f9361387be9523a75)

### Example Rollup interaction diagram
*Shown below is a high level flow of how proxy is used across a rollup stack by different network roles

> (i.e, sequencer, verifier). Any rollup node using an eigenda integration who wishes to sync directly from the parent

Reviewer 2: ```suggestion
(i.e, sequencer, verifier). Any rollup node using an eigenda integration who wishes to sync directly from a parent
```

Author: [41bd2fe](https://github.com/Layr-Labs/eigenda-docs/pull/66/commits/41bd2fe920ff044c79e113a8d247d69b9424ca30)

chain inbox or a safe head must run this service to do so.*

> ![Proxy usage diagram](/img/eigenda/eigenda-proxy-usage-diagram.png)

Reviewer: image link seems broken

Author: [ce5d944](https://github.com/Layr-Labs/eigenda-docs/pull/66/commits/ce5d944dc450ddee1ec6aabf799915fe728d765e)

### Usage
Different actors in the rollup topology will have to use proxy for communicating with EigenDA in the following ways:
- **Rollup Sequencer:** posts batches to proxy and submits accredited DA certificates to batch inbox
- **Rollup Verifier Nodes:** read batches from proxy to update a local state view (*assuming syncing from parent chain
directly)*
- **Prover Nodes:** both rollup types (i.e, optimistic, zero knowledge) will have some way of deriving child chain
state from the parent's inbox for the purpose of generating child --> parent bridge withdraw proofs. These "proving
pipelines" will also read from proxy as well; either for settling disputes in optimistic rollups with working fraud
proofs or for generating zero knowledge proofs attesting to the validity of some batch execution. 

*E.g, In Arbitrum there is a `MakeNode` validator that posts state claims to the parent chain's rollup assertion chain.

> In the event of a challenge, both asserter/challenger players will have to pre-populate their local pre-image stores

Reviewer 2: ```suggestion
In the event of a challenge, both asserter/challenger players will have to pre-populate their local preimage stores
```

Author: [14da642](https://github.com/Layr-Labs/eigenda-docs/pull/66/commits/14da642c388c01bc233306e5249d823855030fbb)

> with batches read from the proxy to compute the WAVM execution traces that they will bisect over.*

Reviewer 2: ```suggestion
with batches read from the proxy to compute the WAVM execution traces necessary for dispute resolution.*
```

Author: [47e7a98](https://github.com/Layr-Labs/eigenda-docs/pull/66/commits/47e7a98bc6784fe2b5fa90b163da1c5bd89265a4)
